watchman.c

```diff
@@ -794,17 +825,17 @@ watchman_clock(struct watchman_connection *conn,
         return NULL;
     }
 
-    json_t *obj = watchman_read(conn, error);
-    if (!obj) {
+    proto_t obj = watchman_read(conn, error);
```

Comment:
would it be possible to do two commits: one to virtualize json_-> proto_ and a second to add bser?  I think that might make this easier to read.